src/perception/lane_detection/src/lane_detection_node.cpp

```diff
@@ -0,0 +1,175 @@
+#include <chrono>
+#include <filesystem>
+#include <functional>
+#include <memory>
+#include <string>
+
+#include <cv_bridge/cv_bridge.h>
+#include <sensor_msgs/msg/compressed_image.hpp>
+#include <sensor_msgs/msg/image.hpp>
+#include "image_transport/image_transport.hpp"
+#include "rclcpp/rclcpp.hpp"
+
+#include "lane_detection_msgs/msg/lane_detection.hpp"
+#include "lane_engine.h"
+#include "std_msgs/msg/string.hpp"
+
+#include <opencv2/opencv.hpp>
+
+#include "common_helper_cv.h"
+#include "image_processor.h"
+
+using namespace std::chrono_literals;
+
+/*** Macro ***/
+#define WORK_DIR RESOURCE_DIR
+
+int count = 0;
+
+class LaneDetectionNode : public rclcpp::Node {
+ public:
+  LaneDetectionNode() : Node("lane_detection"), count_(0) {
+    this->declare_parameter<std::string>("camera_topic", "/CAM_FRONT/image_rect_compressed");
+    this->declare_parameter<std::string>("publish_vis_topic", "/CAM_FRONT/lanes_viz");
+    this->declare_parameter<std::string>("publish_lanes_topic", "/CAM_FRONT/lanes");
+    this->declare_parameter<bool>("save_images", false);
+    this->declare_parameter<std::string>("save_dir", "/tmp");
+    this->declare_parameter<bool>("publish_source_image", false);
+
+    this->get_parameter("camera_topic", camera_topic);
+    this->get_parameter("publish_vis_topic", publish_vis_topic);
+    this->get_parameter("publish_lanes_topic", publish_lanes_topic);
+    this->get_parameter("save_images", save_images);
+    this->get_parameter("save_dir", save_dir);
+    this->get_parameter("publish_source_image", publish_source_image);
+
+    if (!std::filesystem::exists(save_dir)) {
+      std::filesystem::create_directories(save_dir);
+    }
+
+    RCLCPP_INFO(this->get_logger(), "Subscribing to camera topic: %s", camera_topic.c_str());
+
+    subscription_ = this->create_subscription<sensor_msgs::msg::Image>(
+        camera_topic, 10,
+        std::bind(&LaneDetectionNode::image_callback, this, std::placeholders::_1));
+
+    image_pub_ = this->create_publisher<sensor_msgs::msg::Image>(publish_vis_topic, 10);
```

Comment:
Is this only meant for debugging? if so, only init and run this for debugging purposes

Reply:
meant for visualization, similar to how we do it for camera object detection node